### ISSUE TITLE
🐛 BUG: CSS scoped file is generated and lose reference to other files

### ISSUE THREAD

**Issue by Hanawa02** [snowpackjs/astro#445]
### Bug Report: Quick Checklist

- [x] I am on the latest version of Astro & all plugins.
- [x] I am on an ESM Node.js version v14.16.1
- [x] I use package manager: npm.
- [x] I run Astro on OS: Windows.

### Describe the bug

When building (`npm run build`) the CSS referenced as a stylesheet link in the layout is considered as scoped (not sure if this is a bug, but certainly seems weird since it's already the top level), and ends in the `/_astro/` folder, the problem is that this css imports other css files that were, before the build, in the same folder, but after the build are not scoped together.

In resume: 
Before build:
`public/index.css`
`public/theme.css` (imported in the index.css as `@import './theme';`)
`public/theme.css` (imported in the index.css as `@import './code';`)

After build:

`_astro/common-rk2vf.css`
`theme.css` (imported in the common-rk2vf.css as `@import './theme';`)
`theme.css` (imported in the common-rk2vf.css as `@import './code';`)

And the imports return a 404 because the files are searched within the `_astro` folder.

(The index.css is also deployed in the root folder, but no reference is made to it)

### Steps to reproduce

1. `npm init astro` using template docs
2. `npm install`
3. `npm run build` (or you can check out this [repo](https://github.com/Hanawa02/astro-docs/tree/css-issue) 
4. deploy `/dist` files (you can find some live ones [here](http://astro-issue-01.laura-melo.com/)
5. Styles not applied, 404 as described above

### Expected behavior

The css file used in the layouts should not be scoped **OR** the css imported in the scoped files should also be scoped and deployed keeping the same path structure.

**Comment by FredKSchott** [comment]
Yup, related to #454.

If a lot of people are hitting this, I'm happy to champion figuring this out sooner rather than later. Will definitely need everyones feedback. Let me get a proposal together (in the meantime, I outline a rough summary of the system that I'd like to move to here, feel free to comment on that: https://github.com/snowpackjs/astro/issues/454#issuecomment-862953083)

**Comment by Hanawa02** [comment]
@FredKSchott let me know if I can help somehow, or if there is anything I can read/learn in the meantime to be able to help.

**Comment by FredKSchott** [comment]
Awesome, will definitely ping you when I do and would love your help/feedback in the discussion. I only say that I want to kick it off because I'm not sure if the other contributors are on board with the solution outlined in #454, so I want to introduce that approach and get feedback.

**Comment by drwpow** [comment]
Thanks for filing @Hanawa02. Agree that this is unexpected and a bug. But also from our perspective, this is a feature add as this has to fit into bundling, and the solution can be a little tricky—either we inject that `@import`ed file in the wrong place and it breaks your cascade, or we inject it everywhere and you have duplicated styles all over the place (or worse—it recurses and your styles explode in scope). There is a solution, but it‘ll take us a little time.

As a temporary workaround, I’d suggest using [Sass Modules](https://css-tricks.com/introducing-sass-modules/) which will yield the same result but which works today. At least until we have a good solution for bundling `@import`s without completely mangling your styles.

**Comment by Hanawa02** [comment]
@drwpow should I fill another issue to update the docs template using the sass module or just open a PR?

**Comment by drwpow** [comment]
@Hanawa02 a PR would be great if you have time!

**Comment by Hanawa02** [comment]
@drwpow Added to my to-dos for astro.

**Comment by smth** [comment]
A tangentially related comment - I had just ran into a CSS 404 problem, but for me it was due to trying to publish on GitHub Pages, and GitHub Pages ignoring underscored folders by default (I believe adding a `.nojekyll` file should resolve that, but it wasn't working for me for some reason). So if it *is* the intention to link CSS from the `_astro` folder, I guess I would question the naming of it (though I admit most of the problem here is GitHub not behaving as I expected).

**Comment by jonathantneal** [comment]
While I cannot confirm the whole site works (as it relies on Snowpack aliases), I see that the CSS files in public need to reference the correct files. When loading the page with that file (and translating the `snowpack.config.js` file to `vite.config.js`, it appears to be working!

https://stackblitz.com/edit/github-qyweqx?file=public%2Findex.css

**Comment by natemoo-re** [comment]
Following up on Jonathan's comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
